partner_validation_purchase/models/purchase_order.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# © 2018 Akretion
+# © 2018 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models, api, _
+from odoo.exceptions import Warning
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.multi
+    def button_confirm(self):
+        restricted_partners = self.mapped('partner_id.commercial_partner_id').\
+            filtered(lambda p: p.supplier and p.supplier_state != 'approved').\
+            mapped('name')
+        if restricted_partners:
+            raise Warning(_("The supplier ​%s has to be approved"
+                            " in order to confirm this purchase order." % str(restricted_partners)))
```

Comment:
On ne veut pas voir les crochets ouvrants et fermants "[". On veut seulement le nom du partenaire.

![image](https://user-images.githubusercontent.com/27902736/49031610-78cefa00-f178-11e8-94e3-8284a1d8c100.png)

The supplier Ready Mat has to be approved...

ps: sur le screenshot, je suis en Odoo v12, mais le travail était bien à faire en v11 ;)